lib/index.js

```diff
@@ -217,8 +221,14 @@
 
           if (originCallback) {
             originCallback(req.headers.origin, function (err2, origin) {
-              if (err2 || !origin) {
+              if (err2) {
                 next(err2);
+              } else if (!origin) {
```

Comment:
I don't think you need this condition here, as the existing code takes care of this logic. The only change to fix is just removing `!origin`

Reply:
Not really, [this branch](https://github.com/expressjs/cors/blob/master/lib/index.js#L41) will put `'*'` there ...
But it could be probably removed also from there.

Reply:
If that is true, that is also a bug... as false is a valid value for the origin even without the function https://github.com/expressjs/cors#configuration-options

Reply:
Removed.

Reply:
Oh, I see where the false logic is located, which is prior to there. Just copy that false logic from the static options for this. Sorry I am on mobile and wasn't able to grab a link, but the gist is the function return here should work identical to if it was a statix value.